packages/ifpack2/doc/UsersGuide/options.tex

```diff
@@ -191,6 +191,23 @@ \section{Block relaxation}\label{s:block_relaxation}
     {See~\ref{s:relaxation}.}
 \cccc{relaxation: backward mode}
     {See~\ref{s:relaxation}. Currently has no effect. }
+\ccc{block relaxation: decouple dofs}
+    {bool}
+    {false}
+    {Whether to separate blocks according to the different degrees of
+     freedom (PDEs) at each node. This assumes that dofs/node is constant
+     throughout the matrix. Each block will have the same sparsity
+     pattern as the mesh graph's corresponding diagonal block.
+     For example, when using a line partitioner this
+     enables the use of the tridiagonal container even if the matrix's
+     bandwidth is greater than 3.
+     Decoupling is the line smoothing behavior used in ML.}
```

Comment:
Well... I actually never tried making it the default. I was worried that downstream tests might have "gold" solutions that would break from this change. I will try this on my local build and report back here. If there are no failures, this could become the default, although users will have to know that they should change from Banded/Dense to TriDi for their line smoothing.

Reply:
OK, it doesn't break any downstream tests. I'm pushing a commit that does make this the default. Hopefully this will make it easier for applications to compare ML/MueLu results in the future.

Reply:
Albany might care, but if decoupling is almost always better, we should make that the default.  @csiefer2 , would Albany break with this change?

Reply:
@mhoemmen Whoops, this actually did break some MueLu tests, because Ifpack2 is not aware of dofs/node (called "partitioner: PDE equations") and MueLu previously did not have to set that explicitly. So I think I will revert making decoupling the default. I wouldn't say it's better in general (because line smoothing is only one application of BlockRelaxation - there are several kinds of partitioners that decide what a "block" is). It would have been kind of nice to make it the default because Albany expected it to behave like ML, but I don't want to force users to set another parameter with the DOFs.

Reply:
What does this mean, "Ifpack2 is not aware of dofs/node"?  Is this a ParameterList verification issue?  Why is MueLu setting parameters that some Ifpack2 solvers don't know about?

Reply:
I meant that Ifpack2 doesn't know how many dofs/node there are unless "partitioner: PDE equations" is set. The parameter itself is verified and listed in Ifpack2_Parameters.cpp and BlockRelaxation::getValidParameterList().

MueLu always knows the dofs/node and stores it as a level attribute. If MueLu (or anything else) wants to use decoupled BlockRelaxation, it needs to set "partitioner: PDE equations" explicitly.

Reply:
I see -- thanks Brian!